Booking.cs

```diff
@@ -0,0 +1,206 @@
+            string searchOption = Console.ReadLine();
+
+            DateTime? newDepartureDate = null;
+
+            if (searchOption.ToLower() == "y")
+            {
+                Console.Write("Enter the new departure date (MM/dd/yyyy): ");
+                if (DateTime.TryParse(Console.ReadLine(), out DateTime inputDate))
+                {
+                    List<Flight> availableFlights = bookingSystem.SearchFlights(flight.Price, flight.DepartureCountry, flight.DestinationCountry, inputDate, flight.DepartureAirport, flight.ArrivalAirport, flight.Class);
+
+                    if (availableFlights.Any())
+                    {
+                        Console.WriteLine("\nAvailable Flights:");
+                        foreach (var availableFlight in availableFlights)
+                        {
+                            Console.WriteLine($"Flight Code: {availableFlight.FlightNum}, Class: {availableFlight.Class}, Price: {availableFlight.Price}, Departure Date: {availableFlight.DepartureDate}");
+                        }
+
+                        Console.Write("Enter the Flight Code of the desired flight: ");
+                        string selectedFlightCode = Console.ReadLine();
+                        var selectedFlight = availableFlights.FirstOrDefault(flight => flight.FlightNum == selectedFlightCode);
+
+                        if (selectedFlight != null)
+                        {
+                            passenger.UpdateBookedFlights(flight, selectedFlight);
+                            flight.NumberOfSeats = flight.NumberOfSeats +1;
+                            selectedFlight.NumberOfSeats=selectedFlight.NumberOfSeats -1;
+                            Console.WriteLine("Booking updated successfully!");
+                        }
+                        else
+                        {
+                            Console.WriteLine("Invalid Flight Number. Booking not updated.");
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("No available flights on the specified date. Booking not updated.");
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Invalid date format. Booking not updated.");
+                }
+            }
+        }
+        public List<BookingDetails> FilterBookings(
+             string flightNum = null,
+             double? minPrice = null,
+             double? maxPrice = null,
+             string departureCountry = null,
+             string destinationCountry = null,
+             DateTime? departureDate = null,
+             string departureAirport = null,
+             string arrivalAirport = null,
+             int? passengerId = null,
+             Flight.FlightClass? flightClass = null)
+        {
+            var filteredBookings = bookings;
+            if (!string.IsNullOrEmpty(flightNum))
```

Comment:
hmm, too many if statements, think of a way to solve the filters problem using relflection